core/src/test/java/org/apache/shiro/realm/AuthorizingRealmTest.java

```diff
@@ -214,6 +228,36 @@ public void testRealmWithRolePermissionResolver()
         assertTrue( realm.isPermitted( pCollection, "other:bar:foo" ) );
     }
 
+    @Test
+    public void testRealmWithEmptyOrNullPermissions() {
+        Principal principal = new UsernamePrincipal("rolePermResolver");
+        PrincipalCollection pCollection = new SimplePrincipalCollection(principal, "testRealmWithRolePermissionResolver");
+
+        AuthorizingRealm realm = new AllowAllRealm();
+        realm.setRolePermissionResolver( new RolePermissionResolver()
+        {
+            public Collection<Permission> resolvePermissionsInRole( String roleString )
+            {
+                Collection<Permission> permissions = new HashSet<Permission>();
+                if( roleString.equals( ROLE ))
+                {
+                    permissions.add( new WildcardPermission( ROLE + ":perm1" ) );
+                    permissions.add( new WildcardPermission( ROLE + ":perm2" ) );
+                    permissions.add( new WildcardPermission( "other:*:foo" ) );
```

Comment:
Looks good!

Couple minor nits:
We should probably add a null/empty string here too?
and maybe a whitespace string to both? `" "` `"\t "`? (that might be covered in existing tests though)

Reply:
@bdemers done :)